lib/std/fs.zig

```diff
@@ -264,7 +264,13 @@ pub const Dir = struct {
         pub const Kind = File.Kind;
     };
 
-    const IteratorError = error{AccessDenied} || os.UnexpectedError;
+    const IteratorError = error{
+        AccessDenied,
+
+        /// WASI-only. This error occurs when the underlying `Dir` file descriptor does
+        /// not hold the required rights to call `fd_readdir` on it.
+        NotCapable,
```

Comment:
Would it be reasonable to make this the same thing as `AccessDenied`? "does not hold required rights" sounds a lot like access denied. It seems to me like it would be reasonable for callsites to handle WASI's ENOTCAPABLE and other systems' EACCESS the same way.

Reply:
Hmm, that's actually a very good observation. I thought it'd be beneficial to have a separate enum value, but now that I think about this, `AccessDenied` is simple to parse as far as the error value goes *and* it keeps the "internals" away from the user (i.e., they don't need to know about `ENOTCAPABLE` at all unless they choose so by going low-level and using WASI syscalls directly). I like it, thanks @andrewrk!

Reply:
OK, I've redone this as suggested (thanks @andrewrk). Changes in 8306826.